library/classes/Prescription.class.php

```diff
@@ -784,10 +784,9 @@ static function prescriptions_factory(
 
         $prescriptions = array();
         $p = new Prescription();
-        $sql = "SELECT id FROM  " . $p->_table . " WHERE patient_id = " .
-                add_escape_custom($patient_id) .
+        $sql = "SELECT id FROM  " . $p->_table . " WHERE patient_id = ?" .
                 " ORDER BY " . add_escape_custom($order_by);
-        $results = sqlQ($sql);
+        $results = sqlQ($sql, array($patient_id));
         while ($row = sqlFetchArray($results)) {
```

Comment:
Without this fix. possible to get all the prescriptions from the database using -
openemr/controller.php?prescription&list&id=1 or 1=1 --

Reply:
agreed. 1 more down :)